### ISSUE TITLE
Doesn't detect parameter destructuring when the parameter has a default value.

### ISSUE THREAD

**Issue by Shakeskeyboarde** [lukeapage/eslint-plugin-destructuring#46]
Destructuring is a warning in all parameters.

```json
{
  "rules": {
    "destructuring/in-params": ["warn", { "max-params": 0 }]
  }
}
```

This doesn't show a warning:

```ts
const createConnectMiddleware = <TArgs extends readonly unknown[]>(
  id: string,
  factory: (...args: TArgs) => (req: IncomingMessage, res: ServerResponse, next: (error?: unknown) => void) => void,
  { timeoutSeconds = 3 }: MiddlewareConnectOptions = {},
): MiddlewareFactory<TArgs> => {
  ...
}
```

This does show a warning:

```ts
const createConnectMiddleware = <TArgs extends readonly unknown[]>(
  id: string,
  factory: (...args: TArgs) => (req: IncomingMessage, res: ServerResponse, next: (error?: unknown) => void) => void,
  { timeoutSeconds = 3 }: MiddlewareConnectOptions,
): MiddlewareFactory<TArgs> => {
  ...
}
```

Difference:

```
{ timeoutSeconds = 3 }: MiddlewareConnectOptions = {},
{ timeoutSeconds = 3 }: MiddlewareConnectOptions
```

**Comment by lo1tuma** [comment]
I’ve just encountered this as well today. I would be happy to work on a PR if this issue gets accepted. WDYT @lukeapage?
